apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCreateLinks.tsx

```diff
@@ -0,0 +1,92 @@
+import { useMemo } from "react";
+import { TFunction } from "i18next";
+import { ContextMenuItemType } from "~/renderer/components/ContextMenu/ContextMenuWrapper";
+import { openURL } from "~/renderer/linking";
+import { safeList } from "LLD/features/Collectibles/utils/useSafeList";
+import { ExternalIdEnum, ItemType } from "LLD/features/Collectibles/types/enum/Links";
+import IconOpensea from "~/renderer/icons/Opensea";
+import IconRarible from "~/renderer/icons/Rarible";
+import IconMagicEden from "~/renderer/icons/MagicEden";
+import { IconsLegacy } from "@ledgerhq/react-ui";
+import { SimpleHashNft } from "@ledgerhq/live-nft/api/types";
+import { NFTMetadata, ProtoNFT } from "@ledgerhq/types-live";
+import { isNFTMetadata } from "../utils/typeGuardsChecker";
+import { createOrdinalExplorerUrl } from "../utils/createOrdinalExplorerUrl";
+
+const createLinks = (
+  t: TFunction,
+  links: Record<string, string> | SimpleHashNft["collection"]["marketplace_pages"] | undefined,
+  isSimpleHash: boolean,
+  explorerLink?: string,
+): ContextMenuItemType[] => {
+  if (isSimpleHash) {
+    const marketplacePages = Array.isArray(links) ? links : [];
+    return safeList([
+      ...marketplacePages
+        .filter(link => link.marketplace_id === ExternalIdEnum.MAGICEDEN)
+        .map(link => ({
+          id: link.marketplace_id,
+          label: t("NFT.viewer.actions.open", { viewer: link.marketplace_name }),
+          Icon: IconMagicEden,
+          type: ItemType.EXTERNAL,
+          callback: () => openURL(link.nft_url),
+        })),
+
+      ...(marketplacePages.length > 0 ? [{ id: "sep2", type: "separator", label: "" }] : []),
+      explorerLink && {
```

Comment:
if `explorerLink` is falsy, you will get an unwanted `false` value in your array